connectors/douban.fm.js

```diff
@@ -1,22 +1,14 @@
 'use strict';
-
-Connector.playerSelector = '.player-wrapper';
-
-Connector.trackArtSelector = '.playing-cover img';
-
-Connector.getTrack = () => {
-	return $('.player-wrapper .middle>div:nth-child(2) a').first().text();
-};
-
-Connector.getArtist = () => {
-	return $('.player-wrapper .link.artist-name').first().text();
-};
-
-Connector.getDuration = () => {
-	let text = $('.time').text();
-	return Util.stringToSeconds(text.substring(1));
-};
-
-Connector.isPlaying = () => {
-	return $('.player-wrapper .buttons .icon-pause').length > 0;
-};
+const update = (attrs) => {
+	for (let key in attrs) {
+		Connector[key] = attrs[key];
+	}
+};
+update({
+	playerSelector: '.player-wrapper',
+	trackArtSelector: '.playing-cover img',
+	trackSelector: '.middle>div:nth-child(2)>a',
+	artistSelector: '.middle>div:nth-child(3) a',
+	playButtonSelector: '.icon-play',
+	getCurrentTime: () => -Util.stringToSeconds($('.time').text().substring(1)),
```

Comment:
The controller now receives negative duration.

Reply:
That's the played time of the counting down style

Reply:
The former one assigns the absolute value to the duration, continuously triggering onStatusChanged events.

Reply:
Yes, that was a bug.